code/model/modelread.cpp

```diff
@@ -6079,6 +6086,14 @@ uint align_bsp_data(ubyte* bsp_in, ubyte* bsp_out, uint bsp_size)
 			memcpy(&bsp_chunk_size, bsp_in + 4, 4);
 		}
 
+		//Chunk size validation, if fails change it to copy the remaining data in chain
+		auto max_size = end - bsp_in;
+		if (bsp_chunk_size < 0 || bsp_chunk_size > max_size) {
+			
+			mprintf(("Invalid BSP Chunk size detected during BSP data align: Chunk Type: %d, Chunk Size: %d, Max Size: %d\n", bsp_chunk_type, bsp_chunk_size, static_cast<uint>(max_size)));
```

Comment:
This should be a warning, it's a fairly serious problem the modder should know about.

Reply:
changed to a warning.